File: projeto/back/mais-consulta/src/main/java/com/bandtec/mais/consulta/infra/database/DataSourceConfig.java
package com.bandtec.mais.consulta.infra.database;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jdbc.datasource.DriverManagerDataSource;

import javax.sql.DataSource;
import java.util.Properties;

//@Configuration
public class DataSourceConfig {

//
//    @Value("${database.url}")
//    String URL;
//
//    @Value("${database.username}")
//    String USERNAME;
//
//    @Value("${database.password}")
//    String PASSWORD;
//
//    @Value("${database.driver}")
//    String DRIVER;
//
//    @Value("${database.schema}")
//    String DATABASE;
//
//    @Bean
//    public DataSource dataSource() {
//        DriverManagerDataSource dataSource = new DriverManagerDataSource();
//        dataSource.setSchema(DATABASE);
//        dataSource.setCatalog(DATABASE);
//        dataSource.setDriverClassName(DRIVER);
//        dataSource.setUrl(URL);
//        dataSource.setUsername(USERNAME);
//        dataSource.setPassword(PASSWORD);
//
//        return dataSource;
//    }
//
//    @Bean
//    public Properties additionalProperties() {
//        Properties props = new Properties();
//        props.setProperty("hibernate.dialect", "org.hibernate.dialect.SQLServer2012Dialect");
//        props.setProperty("hibernate.show_sql", "true");
//        props.setProperty("hibernate.hbm2ddl.auto", "update");
//
//        return props;
//    }

}

File: projeto/back/maisconsulta/src/main/kotlin/com/bandtec/maisconsulta/infra/database/DataSourceConfig.kt
package com.bandtec.maisconsulta.infra.database

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.datasource.DriverManagerDataSource
import java.util.*
import javax.sql.DataSource

@Configuration
class DataSourceConfig {
    @Value("\${database.url}")
    var URL: String? = null

    @Value("\${database.username}")
    var USERNAME: String? = null

    @Value("\${database.password}")
    var PASSWORD: String? = null

    @Value("\${database.driver}")
    var DRIVER: String? = null

    @Value("\${database.schema}")
    var DATABASE: String? = null

    @Bean
    fun dataSource(): DataSource {
        val dataSource = DriverManagerDataSource()
        dataSource.schema = DATABASE
        dataSource.catalog = DATABASE
        dataSource.setDriverClassName(DRIVER!!)
        dataSource.url = URL
        dataSource.username = USERNAME
        dataSource.password = PASSWORD
        return dataSource
    }

    // Define o hibernate como implementacao do JPA
    @Bean
    fun additionalProperties(): Properties {
        val props = Properties()
        props.setProperty("hibernate.dialect", "org.hibernate.dialect.SQLServerDialect")
        props.setProperty("hibernate.show_sql", "true")
        props.setProperty("hibernate.hbm2ddl.auto", "update")
        return props
    }
}